triggers.go

```diff
@@ -45,6 +46,24 @@ func (s SignalTriggerSequence) Run(info ProcessInfo, sig syscall.Signal) (skipSi
 	return
 }
 
+// SignalTriggerID is the numerical representation of a signal trigger.
+type SignalTriggerID int
```

Comment:
maybe it makes sense to do this by string id and then have a registry (like the one in amboy for jobs/dependencies, or in evergreen for commands,) mapping them to factories rather than needing to know the integer...

also I'm not sure that iota+1 will work...

Reply:
I changed this to use a registry.